extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java

```diff
@@ -935,6 +939,44 @@ private void handleReturn(ClassInfo restClientInterface, String defaultMediaType
                 genericReturnType = createGenericTypeFromParameterizedType(methodCreator, paramType);
             }
         }
+        Integer continuationIndex = null;
+        //TODO: there should be an SPI for this
+        if (returnCategory == ReturnCategory.BLOCKING) {
+            try {
+                Thread.currentThread().getContextClassLoader().loadClass(UNI_KY.toString());
+            } catch (ClassNotFoundException e) {
+                //TODO: make this automatic somehow
+                throw new RuntimeException("Suspendable rest client method" + jandexMethod + " is present on class "
```

Comment:
for users, it may be a bit weird to add mutiny-kotlin. It might be a bit better if we had a dedicated extension for kotlin and rest client.  With it, the check could be based on e.g. on capabilities

But creating a whole extension only for this seems a bit too much...

Reply:
It's not clear to me where we check that this is suspendable?